fastembed/text/text_embedding.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional, Union, Iterable, List, Dict, Any, Type
+
+import numpy as np
+
+from fastembed.text.e5_onnx_embedding import E5OnnxEmbedding
+from fastembed.text.jina_onnx_embedding import JinaOnnxEmbedding
+from fastembed.text.onnx_embedding import OnnxTextEmbedding
+from fastembed.text.text_embedding_base import TextEmbeddingBase
+
+
+class TextEmbedding(TextEmbeddingBase):
+    EMBEDDINGS_REGISTRY: List[Type[TextEmbeddingBase]] = [
```

Comment:
I'd prefer a flag structure over this more confusing list for the registry. E.g. if ONNXTextEmbedding is just FlagEmbedding models, we should add a separate class for Sentence Transformers?

Reply:
could you please elaborate?

Reply:
Is ONNXTextEmbedding specific to FlagEmbedding models? And we'd need to add a new one for GTE (models are here: https://huggingface.co/Qdrant) and Sentence Transformers? 

They need the same code for all 3 implementation though — so I'm not sure if repeated code is better?

Reply:
If it requires small changes in the inference, I would make it inherited from OnnxTextModels, like this:

```
class GTEOnnxTextModels(OnnxTextModels):
  ...
```

I they don't require any changes in the inference, I would just add them to the list